sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -134,6 +145,32 @@ def fit(self, X, y=None):
 
         return self
 
+    def add_element(self, f, v, feature_names, vocab, fitting=True,
+                    transforming=False, indices=None, values=None):
+        if isinstance(v, six.string_types):
+            feature_name = "%s%s%s" % (f, self.separator, v)
+            v = 1
+        elif isinstance(v, Number) or (v is None):
+            feature_name = f
+        elif isinstance(v, Iterable) and not isinstance(v, Mapping):
+            for vv in v:
+                self.add_element(f, vv, feature_names, vocab,
+                                 fitting, transforming, indices, values)
```

Comment:
I'm now thinking that this solution is too general. Lists of numeric values don't make sense, and I don't think we benefit from expanding nested lists. So rather than recursion, please just handle this case specially as a variant of the string value case, and raise an error if any elements of the iterable are not string-like